ambry-router/src/main/java/com.github.ambry.router/TtlUpdateOperation.java

```diff
@@ -240,7 +240,13 @@ private void cleanupExpiredInflightRequests() {
   private void processServerError(ReplicaId replica, ServerErrorCode serverErrorCode, int correlationId) {
     switch (serverErrorCode) {
       case No_Error:
+      case Blob_Already_Updated:
```

Comment:
Looks like there are multiple result that`Blob_Already_Updated` can return: no_error, not_allowed, or forbidden(since update is only supporting one time change).
Just curious what is thought behind the logic `Blob_Already_Updated` returns `No_Error`? 
(I am OK with any of them. )

Reply:
BTW, when is the case that `StoreErrorCodes.Update_Not_Allowed`?

Reply:
@zzmao Will come in the next change on the server side but `StoreErrorCodes.Update_Not_Allowed` happens if the blob has already expired or is "close" to expiry when the TTL update is received.

`Blob_Already_Updated` is a success because the operation as asked has been performed already (the PUT HTTP method is expected to be idempotent btw). It is similar to the reason why we return a success on DELETE of a deleted blob (`DeleteOperation` has similar code where it treats `No_Error` and `Blob_Deleted` the same way)

Reply:
The practical reason is this
1. Client issues TTL update - 1 server acks success, 2 time out. Quorum not reached so op is a failure
2. Client retries TTL update - server that is already updated says so, 1 more server ack successfully this time and third one times out. In effect, the quorum has been reached (2 successful acks). If we don't treat the "already updated" case as a success, then this op will fail again and can never succeed anymore (although the TTL update is actually done)

Reply:
Got it. Thanks!!

Reply:
Will merge once trivis check done.